libsolidity/interface/CompilerStack.cpp

```diff
@@ -1285,12 +1285,30 @@ void CompilerStack::generateEVMFromIR(ContractDefinition const& _contract)
 	//cout << yul::AsmPrinter{}(*stack.parserResult()->code) << endl;
 
 	// TODO: support passing metadata
-	auto result = stack.assemble(yul::AssemblyStack::Machine::EVM);
-	compiledContract.object = std::move(*result.bytecode);
-	// TODO: support runtimeObject
-	// TODO: add EIP-170 size check for runtimeObject
+	// TODO: use stack.assemble here!
+	yul::MachineAssemblyObject init;
+	yul::MachineAssemblyObject runtime;
+	std::tie(init, runtime) = stack.assembleAndGuessRuntime();
+	compiledContract.object = std::move(*init.bytecode);
+	compiledContract.runtimeObject = std::move(*runtime.bytecode);
 	// TODO: refactor assemblyItems, runtimeAssemblyItems, generatedSources,
 	//       assemblyString, assemblyJSON, and functionEntryPoints to work with this code path
+
+	// Throw a warning if EIP-170 limits are exceeded:
+	//   If contract creation initialization returns data with length of more than 0x6000 (214 + 213) bytes,
+	//   contract creation fails with an out of gas error.
+	if (
+		m_evmVersion >= langutil::EVMVersion::spuriousDragon() &&
+		compiledContract.runtimeObject.bytecode.size() > 0x6000
+	)
+		m_errorReporter.warning(
+			9609_error,
+			_contract.location(),
+			"Contract code size exceeds 24576 bytes (a limit introduced in Spurious Dragon). "
+			"This contract may not be deployable on mainnet. "
+			"Consider enabling the optimizer (with a low \"runs\" value!), "
```

Comment:
```suggestion
			"Consider enabling the optimizer (with a low \"optimize-runs\" value!), "
```

Reply:
right?

Reply:
The `runs` is the generic term we have. This message is also a copy of the one present above in `compileContract`.